src/esm_runscripts/workflow.py

```diff
@@ -330,7 +330,13 @@ def init_total_workflow(config):
         config["general"]["workflow"]["subjobs"] = prepcompute
         config["general"]["workflow"]["subjobs"].update(compute)
         config["general"]["workflow"]["subjobs"].update(tidy)
-
+    else:
+        if not "prepcompute" in config["general"]["workflow"]["subjobs"]:
+            config["general"]["workflow"]["subjobs"].update(prepcompute)
+        if not "compute" in config["general"]["workflow"]["subjobs"]:
+            config["general"]["workflow"]["subjobs"].update(compute)
+        if not "tidy" in config["general"]["workflow"]["subjobs"]:
+            config["general"]["workflow"]["subjobs"].update(tidy)    
```

Comment:
I'm a bit puzzled about this, because we have 2 other offline coupled setups and we never found the problem you are describing. The fix looks totally reasonable to me, but the fact that you need it and others didn't makes me wonder what has been done differently. Can you share your runscripts and branch of ESM-Tools that you were using?

Reply:
Hi @mandresm , thanks for your quick response on this. In fact, I was wondering the same, since I followed the documentation and the example for the pism-vilma coupling. Maybe some who is running those 2 setups you mentioned could help here?

I am using the branch `feature/cryogrid` which I just updated.

The runscripts can be found in `/albedo/home/jnitzbon/scripts/runscripts/awiesm-cryogrid` . Below, I copied the sections related to the workflow manger:

"General" runscript:
```
general:
    iterative_coupling: True
    base_dir: /albedo/work/user/jnitzbon/experiments/awiesm-cryogrid-2.1v1.0.0/
    use_venv: False
    coupling_script_dir: /albedo/home/jnitzbon/git/esm_tools/cryogrid-1.0.0/coupling/

model1:
    setup_name: awiesm
    runscript: runscript_awiesm_coupled_albedo.yaml
    chunk_size: 1
    chunk_unit: "years"

model2:
    setup_name: cryogrid
    runscript: runscript_cryogrid_coupled_albedo.yaml
    chunk_size: 1
    chunk_unit: "years"
```
AWI-ESM runscript:
```
[...]
awiesm:
    [...]
    workflow:
        next_run_triggered_by: couple_out
        subjobs: 
            couple_in:
                nproc: 1
                batch_or_shell: shell
                order_in_cluster: sequential
                run_before: prepcompute
                script_dir: ${general.coupling_script_dir} #${cryogrid.model_dir}/coupling/
                script: "scope_receive_echam.sh ${scope.config_file}"
                run_only: first_run_in_chunk
                skip_chunk_number: 1
            couple_out:
                nproc: 1
                batch_or_shell: shell
                order_in_cluster: sequential
                run_after: tidy
                script_dir: ${general.coupling_script_dir} #${cryogrid.model_dir}/coupling/
                script: "scope_send_echam.sh ${scope.config_file}"
                run_only: last_run_in_chunk
[...]
   ```
CryoGrid runscript:
```
[...]
cryogrid:
    [...]
    workflow:
        next_run_triggered_by: couple_out
        subjob_clusters:
            couple_in: couple_in
            couple_out: couple_out
        subjobs: 
            couple_in:
                nproc: 1
                batch_or_shell: shell
                order_in_cluster: sequential
                run_before: prepcompute
                script_dir: ${general.coupling_script_dir} #${cryogrid.model_dir}/coupling/
                script: "scope_receive_cryogrid.sh ${scope.config_file}"
                run_only: first_run_in_chunk
            couple_out:
                nproc: 1
                batch_or_shell: shell
                order_in_cluster: sequential
                run_after: tidy
                script_dir: ${general.coupling_script_dir} #${cryogrid.model_dir}/coupling/
                script: "scope_send_cryogrid.sh ${scope.config_file}"
                run_only: last_run_in_chunk
[...]
```

Reply:
One difference between my runscripts and the ones for the [awicm-pism](https://github.com/esm-tools/esm_tools/tree/release/runscripts/awicm-pism) coupling I noticed is that I added the workflow section within the awiesm section (which is already an online coupled setup) while it is put in the echam and fesom sections (which are standalone models) in the awicm-pism setup.

Reply:
That might be it, because coupled setups sections are merged into `general` during the parsing. I'll have a deeper look later on the day.

Reply:
Hi @jannitzbon, I finally got some time to look into this. I don't have the right permissions for `/albedo/home/jnitzbon/scripts/runscripts/awiesm-cryogrid`. Can you grant them or copy the scripts somewhere where I can access them?

Reply:
I only needed read permissions. However, I'm still not able to access the runscripts:
```shell
$ ls /albedo/work/user/jnitzbon/temp/
ls: cannot access '/albedo/work/user/jnitzbon/temp/': Permission denied
```

Maybe it's because I don't have reading access to the parent directories?

If you prefer it you can copy the runscripts here `/albedo/work/user/mandresm/shared_with_others`